pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -1374,38 +1373,13 @@ func getClusterVersion(cd *hivev1.ClusterDeployment) (string, error) {
 }
 
 func platformAllowsZeroAutoscalingMinReplicas(cd *hivev1.ClusterDeployment) bool {
-	// Since 4.5, AWS, Azure, and GCP allow zero-sized minReplicas for autoscaling
-	if cd.Spec.Platform.AWS != nil || cd.Spec.Platform.Azure != nil || cd.Spec.Platform.GCP != nil {
+	// Zero-sized minReplicas for autoscaling are allowed since OCP:
+	// - 4.5 for AWS, Azure, and GCP
+	// - 4.7 for OpenStack
```

Comment:
musing: other platforms may have been enabled since this was last touched, may be worth looking into.